sinks/splunk/splunk.go

```diff
@@ -137,18 +155,24 @@ func (sss *splunkSpanSink) writeSpan(ctx context.Context, ssfSpan *ssf.SSFSpan)
 	event.SetSourceType(ssfSpan.Service)
 
 	event.SetTime(time.Unix(0, ssfSpan.StartTimestamp))
+	sss.ingest <- event
```

Comment:
The DD sinks uses [container/ring](https://golang.org/pkg/container/ring/) with a configured size to avoid this. We should use that since it's already here and we can avoid the infinite memory!

Reply:
+1 - I was going to suggest this as well!

Reply:
So I didn't use ring.Ring because it seemed quite weird to me:

* To report the spans you have to construct an array anyway (and determining the length of a ring is O(n), so you get to traverse that multiple times).
* Since it's a super naïve double-linked list, it doesn't realllly save memory (I'd wager it consumes more, even, but in smaller units).
* Re-allocating the Ring (like we do in the dd sink) means you get to spend O(initial size) time again making a new one.

Overall, it felt quite ugly to do that over an array which doesn't need munging to fit the existing APIs, and reallocates ~reasonably.

Reply:
So I benchmarked this, and the results are not in exactly `ring.Ring`'s favor. 

I made a benchmark that ingests spans and every 400000 spans, flushes the batch. The two benchmarks below measure the flushing and ingestion respectively. 

First, the array-based batching:
```
BenchmarkBatchFlushing-4          100000             14433 ns/op            3696 B/op         47 allocs/op
BenchmarkBatchIngest-4           1000000              1915 ns/op             338 B/op         13 allocs/op
```

Now the `ring.Ring` batching, with an automatically-growing ring of a constant size (hardcoded to 50k elements):
```
BenchmarkBatchFlushing-4          100000             18965 ns/op            5667 B/op         97 allocs/op
BenchmarkBatchIngest-4           1000000              2388 ns/op             321 B/op         13 allocs/op
```

This isn't extremely much worse, but I think arrays fare a bit better (and I feel the code looks less gross too). WDYT, @aditya-stripe & @cory-stripe?

Reply:
Also, I believe we don't want the traditional ring behavior at all here; either we should attempt to send all received spans to splunk (which the array method does), or we ought to reject them at ingestion time.

Reply:
It seems fine to not use Ring, so long as we set bounds. I agree!

Reply:
Soooo I think it's fine to collect all the spans it can swallow in the flush interval (since they either get submitted or discarded); we probably should find a way to translate pressure back into the ingestion mechanism, but I'm not sure how that would work.

Reply:
I'm fine with discarding at Veneur - getting backpressure to the tailer doesn't do much more for us, and it'd be hard to get all the way back to Envoy (which I don't know if we'd even want?)